### ISSUE TITLE
Do not reload configuration when rancher server is down

### ISSUE THREAD

**Issue by Baklap4** [containous/traefik#2284]
### Do you want to request a *feature* or report a *bug*?
Bug
### What did you do?

With current traefik (v1.4.0)  in combination with rancher server (1.6.10) and nodes i'm having difficulties when the rancher server is down. We thought if the orchestration tool would go down it wouldn't affect the running containers behind traefik. However upon trying to get the configuration from rancher which results in an error (since it's down) traefik does reload his configuration with an empty configuration. Is this intended or not? 

The piece of code for this can be found here (if i'm correct): https://github.com/containous/traefik/blob/v1.4.0/provider/rancher/api.go#L95-L100

If we dig deeper into the method `listRancherEnvironments` we see it correctly logs the error given but still continues with the other methods.

If we take a look to another provider for example `etcd` we see: https://github.com/containous/traefik/blob/v1.4.0/provider/etcd/etcd.go#L25-L27 which returns upon entering an error.

Was this for rancher the intended or not? Considering the logic is not the same as other providers i think this is a bug.

### What did you expect to see?
I expected to see the websites running behind traefik since those containers are still up and traefik had the configuration for the containers before the rancher server went down.

### What did you see instead?
I saw traefik returning a 404 since the configuration changed to not containing any frontends or backends.

log output:
```
time="2017-10-19T14:43:43Z" level=info msg="Skipping same configuration for provider rancher"
time="2017-10-19T14:43:58Z" level=info msg="Skipping same configuration for provider rancher"
time="2017-10-19T14:44:13Z" level=info msg="Skipping same configuration for provider rancher"
time="2017-10-19T14:44:28Z" level=info msg="Skipping same configuration for provider rancher"
time="2017-10-19T14:44:43Z" level=info msg="Skipping same configuration for provider rancher"
time="2017-10-19T14:44:58Z" level=error msg="Cannot get Rancher Environments Get https://roco150.rovecom.nl/v1/environments: dial tcp 193.138.251.150:443: getsockopt: connection refused"
time="2017-10-19T14:44:58Z" level=error msg="Cannot get Provider Services Get https://roco150.rovecom.nl/v1/services?limit=0: dial tcp 193.138.251.150:443: getsockopt: connection refused"
time="2017-10-19T14:44:58Z" level=error msg="Cannot get Provider Services Get https://roco150.rovecom.nl/v1/containers?limit=0: dial tcp 193.138.251.150:443: getsockopt: connection refused"
time="2017-10-19T14:44:58Z" level=info msg="Server configuration reloaded on :443"
time="2017-10-19T14:44:58Z" level=info msg="Server configuration reloaded on :80"
time="2017-10-19T14:45:13Z" level=info msg="Server configuration reloaded on :80"
time="2017-10-19T14:45:13Z" level=info msg="Server configuration reloaded on :443"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
traefik version
Version:      v1.4.0
Codename:     roquefort
Go version:   go1.9.1
Built:        2017-10-16_04:52:07PM
OS/Arch:      linux/amd64
```

**Comment by dbsanfte** [comment]
This is a pretty big pain, Traefik ought to keep going with the last known config if the Rancher server goes down. As it stands it's not very resilient.

**Comment by Baklap4** [comment]
As a matter of fact, if rancher is down it shouldn't even effect all the running containers.. especially since other providers don't have this behavior.

**Comment by wacken89** [comment]
I wrote small crutch
in `server/server.go` between 333 and 336 lines
```go
if configMsg.ProviderName == "rancher" && string(jsonConf) == "{}" {
    log.Infof("Skipping rancher empty config")
} else if configMsg.Configuration == nil || configMsg.Configuration.Backends == nil && configMsg.Configuration.Frontends == nil {
```

**Comment by Baklap4** [comment]
@wacken89 wouldn't it suffice if upon error to return the current rancher item (environment list, serviceList, containerList )?
https://github.com/containous/traefik/blob/v1.4.6/provider/rancher/api.go#L142
```
    if err != nil {
	log.Errorf("Cannot get Rancher Environments %+v", err)
        return environmentList
    }
```

https://github.com/containous/traefik/blob/v1.4.6/provider/rancher/api.go#L159
```
    if err != nil {
	log.Errorf("Cannot get Provider Services %+v", err)
        return servicesList
    }
```

https://github.com/containous/traefik/blob/v1.4.6/provider/rancher/api.go#L176
```
    if err != nil {
	log.Errorf("Cannot get Provider Services %+v", err)
        return containerList
    }
```

**Comment by wacken89** [comment]
@Baklap4 Maybe, but today is may first day with GO, and this fix was critical for me. I wrote it how knew. Guys from Traefik told me how it can be fixed, but i didn't do that for the moment

**Comment by SantoDE** [comment]
As mentioned on slack, if you got questions just shoot @wacken89 . We get that sorted soon :)

**Comment by Ibmurai** [comment]
This is very important, as Rancher will be down when upgrading it. Right now, it's a show stopper for me.

**Comment by ldez** [comment]
Fixed by #2706 in 1.5
